### ISSUE TITLE
Wifi and Mobile data | Not Connection

### ISSUE THREAD

**Issue by bilalbaz1** [videosdk-live/webrtc#15]
I'm running it across different internet connections. I set up a TURN server. It does not work on some internet connections. It usually doesn't work between wifi and mobile data. Has anyone encountered this problem? I've been searching for days but I couldn't find a solution.

(Not device related. My internet connection speed is not slow.)

![IMG_ok](https://github.com/videosdk-live/webrtc/assets/52584017/2b695d84-668e-43d0-b7a4-450df16ff5b0)
![IMG_no](https://github.com/videosdk-live/webrtc/assets/52584017/0a739166-e0bd-439b-a7fc-626542d449e0)

**Comment by harithm918** [comment]
yes. I'm also facing this issue. any solution?

**Comment by bilalbaz1** [comment]
> yes. I'm also facing this issue. any solution?

No. I couldn't find any solution. I also tried it on packages like peerdart [https://pub.dev/packages/peerdart]. The same problem occurs in many cases. It works on some internet connections. It doesn't work on some.

`D/FlutterWebRTCPlugin(19001): onConnectionChangeCONNECTING`
`D/FlutterWebRTCPlugin(19001): onConnectionChangeFAILED`

**Comment by justbendev** [comment]
You need to do some reading on how WebRTC works. Its actually really interesting.
In your specific case you couldn't do NAT Hole Punching 
_Basically create a port forwarding by exploiting firewall state tables_

This demo is using Google Public STUN Servers to help with that as defined in the iceServers list.
If it fails you need a 3rd party server running a TURN to relay packets between your clients.
_Basically a last resort backup when direct P2P fails_

In this demo its located at lib/screens/call_screen.dart line:62

**Comment by guruatdigni** [comment]
any one fixed this issue ?
